### ISSUE TITLE
Add resource to file open for input/output streams

### ISSUE THREAD

**Issue by michael-conway** [DICE-UNC/jargon#221]
Add RESC_NAME_KW, REPL_NUM_KW, advisory lock to correspond to rcDataObjOpen.

**Comment by michael-conway** [comment]
add to IRODSFileSystemAOImpl

```java

@Override
	public int openFile(final IRODSFile irodsFile, final DataObjInp.OpenFlags openFlags) throws JargonException {

		log.info("openFile(final IRODSFile irodsFile,final DataObjInp.OpenFlags openFlags)");

		if (irodsFile == null) {
			throw new JargonException("irodsFile is null");
		}

		String absPath = resolveAbsolutePathGivenObjStat(getObjStat(irodsFile.getAbsolutePath()));

		DataObjInp dataObjInp = DataObjInp.instanceForOpen(absPath, openFlags);

		if (log.isInfoEnabled()) {
			log.info("opening file:" + absPath);
		}

		Tag response = getIRODSProtocol().irodsFunction(IRODSConstants.RODS_API_REQ, dataObjInp.getParsedTags(),
				DataObjInp.OPEN_FILE_API_NBR);

		if (response == null) {
			String msg = "null response from IRODS call";
			log.error(msg);
			throw new JargonException(msg);
		}

		// parse out the response
		int fileId = response.getTag(MsgHeader.PI_NAME).getTag(MsgHeader.INT_INFO).getIntValue();

		log.debug("file id for opened file:{}", fileId);

		return fileId;
	}


```

**Comment by michael-conway** [comment]
closed in #57 as an iRODS issue
